src/main/java/org/springframework/data/elasticsearch/repository/support/SimpleReactiveElasticsearchRepository.java

```diff
@@ -147,7 +160,21 @@ public SimpleReactiveElasticsearchRepository(ElasticsearchEntityInformation<T, I
 	public Mono<Void> deleteById(Publisher<ID> id) {
 
 		Assert.notNull(id, "Id must not be null!");
-		return Mono.from(id).flatMap(this::deleteById);
+		return Flux.from(id)//
```

Comment:
Let's not perform a `collectList` as this way, we're changing behavior and get into the business of buffering requests. We accept `Publisher<ID>` in the method signature to not require users to use Project Reactor's `Mono`, but rather be more open.